### PR TITLE
rclone: extend timeout from 60s to 240s

### LINKED ISSUE #3511: restic with rclone using MEGZ.nz - net/http: request canceled (Client.Timeout exceeded while awaiting headers)
The output of `restic version`
--------------------------
```
$ restic version
restic 0.9.4 compiled with go1.11.6 on linux/amd64
~$ rclone version
rclone v1.56.0
- os/version: debian 10.10 (64 bit)
- os/kernel: 5.10.0-0.bpo.7-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.16.5
- go/linking: static
- go/tags: none
```

How did you run restic exactly?
-------------------------------
```
$ sudo restic -r rclone:mega:/folder_rclone_restic --verbose backup /folder_to_backup
open repository
```
or even
```
$ sudo restic -r rclone:mega:/folder_rclone_restic  -o rclone.args="serve restic --stdio --b2-hard-delete --contimeout 5m --timeout 5m --expect-continue-timeout 5m --verbose" --verbose backup /folder_to_backup
open repository
```
results in 
```
Fatal: unable to open repo at rclone:mega:/folder_rclone_restic: error talking HTTP to rclone: Get http://localhost/file-5577006791947779410: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
internal `rclone serve` and MEGA.nz

Expected behavior
-----------------

Proper backup

Actual behavior
---------------

I got a timeout error (which I was half expecting). Mega is notorious for being slow upon loading up its website, especially with large amounts of data. (In my case, ~2TB). Their website usually hangs at a "Loading" or "Decrypting" screen for 2 mins before showing my files.

Steps to reproduce the behavior
-------------------------------

Do a `restic -r rclone:mega:/ backup /` (successfully backing up 2TB)
Attempt again with the SAME command. (Timeout)

Do you have any idea what may have caused this?
-----------------------------------------------
Doing a google search of ` net/http: request canceled (Client.Timeout exceeded while awaiting headers)` brings up a lot of info about the `net/http` go lib, so it can be rclone (written in go) or restic (also go)
I did a 
```
$ rclone serve restic -v mega:/folder_rclone_restic
2021/09/12 20:16:03 NOTICE: mega root 'folder_rclone_restic': Serving restic REST API on http://localhost:8080/
```
though it takes a minute, it seemed to have no issues. That rules our rclone.

Looking at restic, I think it is this line here : Line 222 of https://github.com/restic/restic/blob/master/internal/backend/rclone/backend.go
```
	// send an HTTP request to the base URL, see if the server is there
	client := &http.Client{
		Transport: debug.RoundTripper(tr),
		Timeout:   60 * time.Second,
	}
```

Only allowing a 60-second timeout.

Do you have an idea how to solve the issue?
-------------------------------------------
Allow for a CLI argument to override the 60s?

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes, by far the fastest and best backup solution out of the multiple I have tested.
I would love to contribute but I know nothing about go and am far too busy to learn it at the moment.
If you'd like, I could test a compiled binary with the proposed changes that _should_ fix this issue. Heck, I will add it to the docs too.

### PR DESCRIPTION
**Pull request by phcreery**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Resolves https://github.com/restic/restic/issues/3511
Timeout error when opening slow rclone backend (ex. MEGA)

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
I initially proposed adding a flag, but that would mean the flag only applies to rclone backend, so I just extended the hard-coded timeout for rclone.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [ ] I'm done! This pull request is ready for review.

### CODE REVIEW

**Comment by phcreery** [comment]
There was no scientific reason for choosing 240s, just an assumption it won't take longer than 4 minutes.

**Comment by phcreery** [comment]
I, too, apologize for the late reply. 
Valid argument and that is a much better implementation you have stated. 
I will look into doing that.

**Comment by phcreery** [comment]
Okay, I added a `-o rclone.timeout=` to the args and documentation.

File: internal/backend/rclone/config.go [hunk 1]
```diff
@@ -12,13 +12,15 @@ type Config struct {
...
 	Remote      string
+	Timeout     uint `option:"timeout"     help:"set a timeout limit (in mins) to wait for rclone to establish a connection (default: 1)"`
 	Connections uint `option:"connections" help:"set a limit for the number of concurrent connections (default: 5)"`
 }
 
 var defaultConfig = Config{
 	Program:     "rclone",
 	Args:        "serve restic --stdio --b2-hard-delete",
 	Connections: 5,
+	Timeout: 1,
```
Comment (MichaelEischer) [inline internal/backend/rclone/config.go:23]: Please run `gofmt`.

File: internal/backend/rclone/config.go [hunk 2]
```diff
@@ -12,13 +12,15 @@ type Config struct {
 	Program     string `option:"program" help:"path to rclone (default: rclone)"`
 	Args        string `option:"args"    help:"arguments for running rclone (default: serve restic --stdio --b2-hard-delete)"`
 	Remote      string
+	Timeout     uint `option:"timeout"     help:"set a timeout limit (in mins) to wait for rclone to establish a connection (default: 1)"`
```
Comment (MichaelEischer) [inline internal/backend/rclone/config.go:15]: Please use `time.Duration`. Then a user can specify `60s` or `1m` which makes clear which unit is used.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]

**Comment by MichaelEischer** [comment]
The tests are currently failing as `internal/backend/rclone/config_test.go` requires an update for the change config data structure.

File: changelog/unreleased/pull-3514 [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Bugfix: timeout error on slow rclone backend
```
Comment (MichaelEischer) [inline changelog/unreleased/pull-3514:1]: I'd change this to:
`Enhancement: Support timeout configurable for rclone backend`

File: changelog/unreleased/pull-3514 [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+
+restic can now be configured with a timeout argument for the `rclone serve` to be ready.
```
Comment (MichaelEischer) [inline changelog/unreleased/pull-3514:3]: What about the following description:
```
A slow rclone backend could cause restic to time out while waiting for the repository to open.
Restic now offers an `-o rclone.timeout` option to make this timeout configurable.
```

Comment (phcreery) [reply changelog/unreleased/pull-3514:3]: That sound better

File: internal/backend/rclone/config.go (hunk 1 above)
Comment (phcreery) [reply internal/backend/rclone/config.go:23]: will do

File: internal/backend/rclone/config.go [hunk 5]
```diff
@@ -12,13 +13,15 @@ type Config struct {
...
-	Connections uint `option:"connections" help:"set a limit for the number of concurrent connections (default: 5)"`
+	Connections uint          `option:"connections" help:"set a limit for the number of concurrent connections (default: 5)"`
+	Timeout     time.Duration `option:"timeout"     help:"set a timeout limit to wait for rclone to establish a connection (default: 1m)"`
 }
 
 var defaultConfig = Config{
 	Program:     "rclone",
 	Args:        "serve restic --stdio --b2-hard-delete",
 	Connections: 5,
+	Timeout:     1,
```
Comment (MichaelEischer) [inline internal/backend/rclone/config.go:24]: that's not 1 minute, but rather 1 nanosecond (after fixing the square seconds above).

File: internal/backend/rclone/backend.go [hunk 6]
```diff
@@ -222,7 +222,7 @@ func newBackend(cfg Config, lim limiter.Limiter) (*Backend, error) {
 	// send an HTTP request to the base URL, see if the server is there
 	client := &http.Client{
 		Transport: debug.RoundTripper(tr),
-		Timeout:   60 * time.Second,
+		Timeout:   time.Duration(cfg.Timeout) * 60 * time.Second,
```
Comment (MichaelEischer) [inline internal/backend/rclone/backend.go:225]: just use a plain `cfg.Timeout` here. At the moment we have square seconds here.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
Please also squash all commits into a single one.

File: internal/backend/rclone/backend.go (hunk 6 above)
Comment (phcreery) [reply internal/backend/rclone/backend.go:225]: Oh yea, good catch

File: internal/backend/rclone/config.go (hunk 5 above)
Comment (phcreery) [reply internal/backend/rclone/config.go]: You are correct, fixed

**Review by MichaelEischer — APPROVED** [review]
LGTM.

I've rebased the PR, squashed the commits and fixed the timeout calculation along with the compilation error.

**Comment by phcreery** [comment]
Thanks!

**Pull request merged**